### DISCUSSION TITLE
Add to the documentation of the Product Editor Development Handbook, how save the new fields?

### DISCUSSION THREAD

**Discussion by racmanuel** [woocommerce/woocommerce#41553] [WooCommerce New Product Editor] [answered]
Add to the documentation of the Product Editor Development Handbook, how save the new fields?  I have not found any information related to this in the documentation.

And what will happen to the metaboxes? Will there be the possibility of showing the content of the metaboxes when it is not a custom field? I have some plugins that show static content. Would it be possible to show the content through hooks?

**Comment by mattsherman** [comment, marked as answer]
Hi @racmanuel 

> Add to the documentation of the Product Editor Development Handbook, how save the new fields? I have not found any information related to this in the documentation.

This should definitely be added to the documentation.

I recently answered a similar question to this on the community Slack ([thread](https://woocommercecommunity.slack.com/archives/C064QAWNDS9/p1700510886124379))...

When using the `property` attribute to link blocks to the underlying product data, the blocks assume/require that the product data object returned from the REST API has those properties on it.

If you wanted to customize/configure the saving and retrieval of custom product properties, you would want to use hooks to accomplish that. The two hooks that are used are:

- `woocommerce_rest_prepare_product_object`
- `woocommerce_rest_insert_product_object`

I’m not finding any documentation on these right now (I’m making a note that we need documentation for these), but you can see where the hooks are applied in https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/rest-api/Controllers/Version2/class-wc-rest-products-v2-controller.php

Note that if you are using product meta data fields via a `property` attribute (something like `meta_data.custom_prop`), those custom meta data fields are automatically saved/retrieved for you. You do not need to use the above mentioned hooks in that case.

> And what will happen to the metaboxes? Will there be the possibility of showing the content of the metaboxes when it is not a custom field? I have some plugins that show static content. Would it be possible to show the content through hooks?

We are still working through how to best handle cases like this. Can you provide us with some examples of the sort of static content that your plugins show (screenshots would be useful)? That will help us determine how to best support this.
